frontend/src/app/[locale]/events/[eventSlug]/surveys/[surveySlug]/responses/summary/FieldSummaryComponent.tsx

```diff
@@ -0,0 +1,137 @@
+import SelectFieldSummaryComponent from "./SelectFieldSummaryComponent";
+import TextFieldSummaryComponent from "./TextFieldSummaryComponent";
+import {
+  Choice,
+  Field,
+  FieldSummary,
+  SelectFieldSummary,
+} from "@/components/SchemaForm/models";
+import { Translations } from "@/translations/en";
+
+interface FieldSummaryComponentProps {
+  translations: Translations;
+  field: Field;
+  fieldSummary: FieldSummary;
+}
+
+// NOTE: if you rename this to FieldSummary, it'll clash with models and graphql-typegen will silently fail
+export default function FieldSummaryComponent({
+  translations,
+  field,
+  fieldSummary,
+}: FieldSummaryComponentProps) {
+  const t = translations.Survey;
+
+  let choices: Choice[] = [];
```

Comment:
I'd probably break down lines 25..62 to a second helper function that could just return `{choices, questions}`.